### PR TITLE
Make the cache files in a child folder of user document folder on Windows.

### PR DESCRIPTION
**Pull request by hxdhttk**
The current location of these cache files may confuse the user since they do not imply any relations to this app, the user may think they are harmful and want to delete them and then find they will be regenerated indeterminately.

Current State:
image

This PR:
image
image

### CODE REVIEW

**Comment by DattatreyaReddy** [comment]
Hi @hxdhttk, 
can you use the `path` package to join the paths instead of hardcoding it? 
so we could use this on all platforms instead of only Windows.

```dart
import 'package:path/path.dart' as path;
  Directory? appDirectory;
  if (!kIsWeb) {
    appDirectory = await getApplicationDocumentsDirectory();
    appDirectory = Directory(path.join(appDirectory.path, 'Sorayomi'));
  } else {
    appDirectory = null; // to avoid initialization error I think
  }
```
 something like above?
 you may need to add path package btw!

**Comment by hxdhttk** [comment]
Hi @DattatreyaReddy,
The PR was updated.

File: lib/main.dart [hunk 1]
```diff
@@ -21,7 +24,27 @@ Future<void> main() async {
...
+
+    await appDirectory.create(recursive: true);
+
+    final cacheFiles = ['dio_cache.hive', 'dio_cache.lock'];
+    for (final cacheFile in cacheFiles) {
+      final oldCacheFilePath = path.join(appDocDirectory.path, cacheFile);
+      final newCacheFilePath = path.join(appDirectory.path, cacheFile);
+
+      if (!(await File(newCacheFilePath).exists())) {
+        await File(oldCacheFilePath).rename(newCacheFilePath);
```
Comment (DattatreyaReddy) [inline lib/main.dart:41]: @hxdhttk it seems like rename may not work all times so we can use copy & delete method as mentioned in [official doc](https://api.flutter.dev/flutter/dart-io/File/rename.html#:~:text=rename%20abstract%20method,-Future%20rename&text=Renames%20this%20file.,File%20for%20the%20renamed%20file.&text=On%20some%20platforms%2C%20a%20rename,and%20then%20remove%20the%20original.)

File: lib/main.dart [hunk 2]
```diff
@@ -21,7 +24,27 @@ Future<void> main() async {
   WidgetsFlutterBinding.ensureInitialized();
   final packageInfo = await PackageInfo.fromPlatform();
   final sharedPreferences = await SharedPreferences.getInstance();
-  var appDirectory = (kIsWeb) ? null : await getApplicationDocumentsDirectory();
+
+  Directory? appDirectory;
+  if (!kIsWeb) {
+    final appDocDirectory = await getApplicationDocumentsDirectory();
+    appDirectory = Directory(path.join(appDocDirectory.path, 'Sorayomi'));
```
Comment (DattatreyaReddy) [inline lib/main.dart:31]: you can use `packageInfo.appName` as you did in prev [commit](https://github.com/Suwayomi/Tachidesk-Sorayomi/pull/216/commits/3d38941f3a77cd3b7bae4a01e2cc5413e8172ace#diff-e61eb31d013d12616f5532636a88cfa63631dda8f7829e5424e68542214d1608R28)

**Review by DattatreyaReddy — CHANGES_REQUESTED** [review]
Thank you for Raising the PR

**Pull request merged**